### ISSUE TITLE
Polish translation

### ISSUE THREAD

**Issue by siema** [fafranco82/swdestinydb#21]
I'd like to help with getting Polish translation up and running :)
I've read #18 about what has to be done, and from what I see, I can start working on cards right away, but site translation needs a tiny bit of preparations first, right?

**Comment by fafranco82** [comment]
Yes, but I've just uploaded the needed files. You can edit `messages.pl.yml` now ;)

Also, I have been mailed by a user that also want to help with the polish translation, but he don't know how to use github. Maybe he can help you. May I give him some contact info of you?

**Comment by siema** [comment]
Of course :)

**Comment by prozz** [comment]
we are organizing translation effort on game's facebook group. including the user who mailed u @fafranco82. there is healthy mix of users with github knowledge and ability and will to translate too. hope for fast progress :)

**Comment by siema** [comment]
Yeah, I first wrote the ticket, then checked local Polish group on FB :P

**Comment by fafranco82** [comment]
I'm glad to hear that :)

**Comment by fafranco82** [comment]
Hi there,

I've just pushed a big update concerning translations. I've changed the format of translation files from YML to XLIFF. This files are easier to maintain because can have extra info and there are tools designed to use this kind of file.

With your translation work in progress, this is the necessary steps to keep your work safe:

* Copy your translation files (`messages.pl.yml`) to another place, because the file was deleted in the repository and will be deleted too in your working copy.
* Pull upstream master and run again `composer install`
* Copy back the `messages.pl.yml` to the same location it was before.
* Run `php app/console translation:extract --config=AppBundle pl`. This will take into account the translations made in `messages.fr.yml` and then transform the file in XLIFF format, renaming it to `messages.pl.xliff`.
* Run `php app/console server:run` and then navigate to `/_trans` path.
* Select `AppBundle`, `messages` and `pl` in the select inputs on the top. You can edit here the translations. They will be automatically saved on the file when the textarea lose focus.

I hope you can understand this change... whenever a new translation key will be needed, it will be easier to translate it in all languages this way.

**Comment by siema** [comment]
No problem, we're still collecting card data and haven't really touched the website itself :P

**Comment by fafranco82** [comment]
FYI, maybe you find some troubles related with this:

https://github.com/fafranco82/swdestinydb/issues/18#issuecomment-263697530

That is the solution to the problem stated in the comment above. Only change `fr` for `pl` in your case.

**Comment by siema** [comment]
I haven't even gotten to that point. At the end of `composer install` and in response to any other command after that, all I get is:
`

  [InvalidArgumentException]
  Credentials must be an instance of Aws\Credentials\CredentialsInterface, an associative array that contains "key", "secret", and an optional "token" key-value pairs, a credentials provid
  er function, or false.


`
I haven't configured it, all AWS-related entries are null in parameters.yml.

**Comment by siema** [comment]
How can I define formatting of numerals? I mean this one:
`{0} There is no cards | {1} One card | ]1,Inf] %count% cards`
In Polish it's kinda weird. Except for 1-20, it depends on the last digit, so I'd need some kind of modulo function in the expression...
Because it goes like:
- 1 karta
- 2..4 karty
- 5..21 kart
- 22..24 karty
- 25..31 kart
and so on

**Comment by fafranco82** [comment]
Reading this: http://symfony.com/doc/current/components/translation/usage.html#pluralization

I supposed you need only to write the 3 different forms separated with `|` and the internally (and using the locale information), it will pick the right form. I guess the order is the "natural order", i.e., the first choice if the first you use when counting, the second choice the second you use, and so on, although you use the "first form" again.

So, try this for the name of cards:

`%count% karta | %count% karty | %count% kart`

In addition to this, you can see explicit interval choices, and, when a number does not match an explicit interval choice, the right non-interval choice will be picked (ignoring explicit interval choices before them).

So, if you wish to say "There is no cards" in Polish instead "0 cards", you can try this:

`{0} Nie ma karty | %count% karta | %count% karty | %count% kart`

Tell me if this was useful :)

PD: I used Google Translator for the zero-cards translation, hehehe

**Comment by siema** [comment]
Did it like you suggested, looks fine, thanks :)

**Comment by fafranco82** [comment]
You're welcome. Which kind of message did you use? Former or latter?

**Comment by siema** [comment]
The latter one, with explicit "0" form

**Comment by siema** [comment]
Aaaand I have it mostly done :)
Though there are some layout bugs, due to longer names in some places. See attached screenshot.
![swdestinydb](https://cloud.githubusercontent.com/assets/537808/20833380/aec64da8-b88f-11e6-97f5-9eeacae332a9.png)

**Comment by fafranco82** [comment]
Yes, I can see the problems with line break, I will fix it later.

About the polish characters... it's curious... it seems that CSS transform function doesn't know what is the capital letter of polish ones... I will investigate about this. What browser are you using? In case this was a browser issue.

**Comment by fafranco82** [comment]
http://jsfiddle.net/6uy09kps/2/

In Chrome works the capitalization, also in Firefox and Chrome....

But not in the site...
![screenshot-swdestinydb com 2016-12-02 13-38-14](https://cloud.githubusercontent.com/assets/5708531/20834292/d80a5254-b894-11e6-99da-1dffbf9bdff6.png)

Maybe is because the font...

**Comment by siema** [comment]
I tested on Chrome and Firefox on Mac

**Comment by fafranco82** [comment]
Yes, it's because the font.... the 'versalitas' style (small letters as capital letters) it is not an CSS transform, but the glyph in the font... and doesn't support capital polish special characters... I need to find a good webfont and use CSS transform.

Meanwhile, you can go on with the translation without worrying about this. I'm going to create a new issue with this.

**Comment by fafranco82** [comment]
With 2a58ac178e7c08a1d259849acae355c7c74e9dbc, this issue is fixed :)

**Comment by siema** [comment]
Alright, sent pull requests for both repos. Although the json-data one is failing CI for some reason....

**Comment by fafranco82** [comment]
There are several new translation keys in the project ;)

**Comment by fafranco82** [comment]
@siema I'm trying a new way for translation, using the site Loco. I'd like to send you an invitation, but I need an email and name...

**Comment by fafranco82** [comment]
I've update the README with the new way of translating the site. You can send me an email (to webmaster@swdestinydb.com) with your name and email you send you back an invitation to the translation tool ;)

**Comment by siema** [comment]
Didn't check Github for few days, I've just send the email :)
